### PR TITLE
Observe idle; save project view state. Closes #3791

### LINKED ISSUE #3791: TopView state isn't saving properly, wrong files opened after crash/forced quit
### Short Summary

I'm not 100% on repro steps for this other than it involves a forced quit/manually kill the process or a crash.  Either the open view state isn't being saved when the rest of the window state gets saved or Komodo recovery code is loading the saved state from somewhere else.  I'm pretty sure it's the latter.

### Steps to Reproduce

Force quit Komodo however your platform supports that
Restart Komodo

### Expected results

Should open with all the files you had open before the crash

### Actual results

Some long gone state is loaded.

### PR DESCRIPTION
**Pull request by ssigwart**

### CODE REVIEW

File: src/prefs/koPrefs.py [hunk 1]
```diff
@@ -1571,6 +1571,7 @@ def __init__(self):
                     getService(components.interfaces.nsIObserverService)
         obsvc.addObserver(self, 'xpcom-shutdown', False)
         obsvc.addObserver(self, 'profile-before-change', False)
+        obsvc.addObserver(self, 'idle', False)
```
Comment (th3coop) [inline src/prefs/koPrefs.py]: As far as I can tell, this shouldn't have an effect.  I've not looked at this before but it looks like idle monitoring is handled by an underlying Mozilla service which is at line 1548.  Did you confirm that this changes anything in `def observe`?  I don't think this should be added.

File: src/chrome/komodo/content/workspace.p.js [hunk 2]
```diff
@@ -649,6 +649,10 @@ this.saveWorkspace = function view_saveWorkspace(saveNow)
                                           thisWindow, mainWindow,
                                           windowWorkspace, saveCoordinates);
         }
+        // Save current project settings
+        var currentProject = ko.projects.manager.currentProject;
+        if (currentProject)
```
[the remark this answers is not in the document]
Comment (ssigwart) [reply src/chrome/komodo/content/workspace.p.js]: Added.

File: src/prefs/koPrefs.py (hunk 1 above)
Comment (ssigwart) [reply src/prefs/koPrefs.py]: I did add debugging to `observer` and never saw an idle `topic`.  However, maybe I didn't wait long enough.  I removed it.

Comment (th3coop) [reply src/prefs/koPrefs.py]: You added the debugging before or after the change? I'd be happy to keep this if you tested before the change and saw no calls then tested after and saw action.  If there was no change let's keep it out.

Comment (ssigwart) [reply src/prefs/koPrefs.py]: Yes.  I added it before and after.  It worked after, but not before, but I think I switched to the terminal to watch the log and it appears that Mozilla doesn't fire idle if you aren't focused on the app.  I realized that later and forgot to retest the original code.  I tested again and it looks like it's not needed.

**Pull request merged**